diff --git a/node_modules/@capacitor/push-notifications/ios/Sources/PushNotificationsPlugin/PushNotificationsHandler.swift b/node_modules/@capacitor/push-notifications/ios/Sources/PushNotificationsPlugin/PushNotificationsHandler.swift
index f3972b6..b6b8549 100644
--- a/node_modules/@capacitor/push-notifications/ios/Sources/PushNotificationsPlugin/PushNotificationsHandler.swift
+++ b/node_modules/@capacitor/push-notifications/ios/Sources/PushNotificationsPlugin/PushNotificationsHandler.swift
@@ -29,7 +29,7 @@ public class PushNotificationsHandler: NSObject, NotificationHandlerProtocol {
             }
         }
 
-        if let optionsArray = self.plugin?.getConfig().getArray("presentationOptions") as? [String] {
+        if let optionsArray = self.plugin?.getConfig().getConfigJSON()["presentationOptions"] as? [String] {
             var presentationOptions = UNNotificationPresentationOptions.init()
 
             optionsArray.forEach { option in
@@ -83,7 +83,14 @@ public class PushNotificationsHandler: NSObject, NotificationHandlerProtocol {
             "subtitle": request.content.subtitle,
             "badge": request.content.badge ?? 1,
             "body": request.content.body,
-            "data": JSTypes.coerceDictionaryToJSObject(request.content.userInfo) ?? [:]
+            "data": request.content.userInfo.reduce(into: JSObject()) { result, item in
+                guard let key = item.key as? String else { return }
+                if let value = item.value as? JSValue {
+                    result[key] = value
+                } else {
+                    result[key] = String(describing: item.value)
+                }
+            }
         ]
     }
 }
diff --git a/node_modules/@capacitor/push-notifications/ios/Sources/PushNotificationsPlugin/PushNotificationsPlugin.swift b/node_modules/@capacitor/push-notifications/ios/Sources/PushNotificationsPlugin/PushNotificationsPlugin.swift
index 9044c74..badb335 100644
--- a/node_modules/@capacitor/push-notifications/ios/Sources/PushNotificationsPlugin/PushNotificationsPlugin.swift
+++ b/node_modules/@capacitor/push-notifications/ios/Sources/PushNotificationsPlugin/PushNotificationsPlugin.swift
@@ -78,11 +78,11 @@ public class PushNotificationsPlugin: CAPPlugin, CAPBridgedPlugin {
         self.notificationDelegateHandler.requestPermissions { granted, error in
             guard error == nil else {
                 if let err = error {
-                    call.reject(err.localizedDescription)
+                    call.unavailable(err.localizedDescription)
                     return
                 }
 
-                call.reject("unknown error in permissions request")
+                call.unavailable("unknown error in permissions request")
                 return
             }
 
@@ -123,7 +123,7 @@ public class PushNotificationsPlugin: CAPPlugin, CAPBridgedPlugin {
      */
     @objc func getDeliveredNotifications(_ call: CAPPluginCall) {
         if !appDelegateRegistrationCalled {
-            call.reject("event capacitorDidRegisterForRemoteNotifications not called.  Visit https://capacitorjs.com/docs/apis/push-notifications for more information")
+            call.unavailable("event capacitorDidRegisterForRemoteNotifications not called.  Visit https://capacitorjs.com/docs/apis/push-notifications for more information")
             return
         }
         UNUserNotificationCenter.current().getDeliveredNotifications(completionHandler: { (notifications) in
@@ -141,11 +141,11 @@ public class PushNotificationsPlugin: CAPPlugin, CAPBridgedPlugin {
      */
     @objc func removeDeliveredNotifications(_ call: CAPPluginCall) {
         if !appDelegateRegistrationCalled {
-            call.reject("event capacitorDidRegisterForRemoteNotifications not called.  Visit https://capacitorjs.com/docs/apis/push-notifications for more information")
+            call.unavailable("event capacitorDidRegisterForRemoteNotifications not called.  Visit https://capacitorjs.com/docs/apis/push-notifications for more information")
             return
         }
-        guard let notifications = call.getArray("notifications", JSObject.self) else {
-            call.reject("Must supply notifications to remove")
+        guard let notifications = call.getArray("notifications", []) as? [JSObject] else {
+            call.unavailable("Must supply notifications to remove")
             return
         }
 
@@ -159,7 +159,7 @@ public class PushNotificationsPlugin: CAPPlugin, CAPBridgedPlugin {
      */
     @objc func removeAllDeliveredNotifications(_ call: CAPPluginCall) {
         if !appDelegateRegistrationCalled {
-            call.reject("event capacitorDidRegisterForRemoteNotifications not called.  Visit https://capacitorjs.com/docs/apis/push-notifications for more information")
+            call.unavailable("event capacitorDidRegisterForRemoteNotifications not called.  Visit https://capacitorjs.com/docs/apis/push-notifications for more information")
             return
         }
         UNUserNotificationCenter.current().removeAllDeliveredNotifications()
